File: back/internal/infra/http-server/contracts/food.go
package contracts

type CreateFoodRequest struct {
	Name           string `json:"name" validate:"required"`
	CaloriesPer100 int    `json:"caloriesPer100" validate:"required,number"`
	Carbs          *int   `json:"carbs" validate:"required,number"`
	Fat            *int   `json:"fat" validate:"required"`
	Protein        *int   `json:"protein" validate:"required"`
}

type FoodResponse struct {
	Name           string `json:"name"`
	CaloriesPer100 int    `json:"caloriesPer100"`
	Carbs          int    `json:"carbs"`
	Id             int    `json:"id"`
	Fat            int    `json:"fat"`
	Protein        int    `json:"protein"`
}

File: front/src/validation/food.ts
import { z } from "zod";

export const CreateFoodSchema = z.object({
  name: z.string(),
  calories: z.number({
    message: "Введите калорийность",
  }),
  carbs: z.number({
    message: "Введите кол-во углеводов",
  }),
  fat: z.number({
    message: "Введите кол-во жиров",
  }),
  protein: z.number({
    message: "Введите кол-во белков",
  }),
});

export const UpdateFoodSchema = CreateFoodSchema.partial().extend({
  id: z.number(),
});

export type CreateFoodData = z.infer<typeof CreateFoodSchema>;
export type UpdateFoodData = z.infer<typeof UpdateFoodSchema>;
